### PR TITLE
using the provider

### PR DESCRIPTION
**Pull request by franciscolopezsancho**
References #11

### CODE REVIEW

File: akka-diagnostics/src/main/scala/akka/diagnostics/StarvationDetector.scala [hunk 1]
```diff
@@ -133,21 +133,22 @@ object StarvationDetector {
...
-  def checkSystemDispatcher(system: ActorSystem, config: StarvationDetectorSettings): Unit =
+  def checkSystemDispatcher(provider: ClassicActorSystemProvider, config: StarvationDetectorSettings): Unit =
     checkExecutionContext(
-      system.dispatcher,
-      Logging(system, classOf[StarvationDetectorThread]),
+      provider.classicSystem.dispatcher,
+      Logging(provider.classicSystem, classOf[StarvationDetectorThread]),
       config,
-      () => system.whenTerminated.isCompleted)
+      () => provider.classicSystem.whenTerminated.isCompleted)
```
Comment (patriknw) [inline akka-diagnostics/src/main/scala/akka/diagnostics/StarvationDetector.scala:152]: we also have in ConfigChecker
```
def reportIssues(system: ExtendedActorSystem)
```

That can also be ClassicActorSystemProvider. I don't know if we need the Extended, but if that is the case you can safely cast it `provider.classicSystem.asInstanceOf[ExtendedActorSystem]`

Comment (franciscolopezsancho) [reply akka-diagnostics/src/main/scala/akka/diagnostics/StarvationDetector.scala:152]: Okay. I'll do that too.

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by franciscolopezsancho** [comment]
I have a question: Here we are instantiating a classic `ActorSystem` https://github.com/akka/akka-diagnostics/pull/34/files#diff-fb7d642c4856f04058e3a54fdd409d2359d7a6f844f53773ebd932ccf1bc03b8L64

This is the `main` used to run it as a java standalone. I don't think this will be a problem as we are scanning the configuration files directly in this case. Would you agree?

**Comment by patriknw** [comment]
> This is the main used to run it as a java standalone.

That is correct. That is only an undocumented convenience.

**Pull request merged**